oxd-server/src/main/java/org/gluu/oxd/server/mapper/RegisterResponseMapper.java

```diff
@@ -0,0 +1,447 @@
+                KeyEncryptionAlgorithm.fromName(respClaims.get(USERINFO_ENCRYPTED_RESPONSE_ALG.toString())) != null) {
+            rpFromRegisterResponse.setUserInfoEncryptedResponseAlg(respClaims.get(USERINFO_ENCRYPTED_RESPONSE_ALG.toString()));
+        }
+
+        if (!Strings.isNullOrEmpty(respClaims.get(USERINFO_ENCRYPTED_RESPONSE_ENC.toString())) &&
+                BlockEncryptionAlgorithm.fromName(respClaims.get(USERINFO_ENCRYPTED_RESPONSE_ENC.toString())) != null) {
+            rpFromRegisterResponse.setUserInfoEncryptedResponseEnc(respClaims.get(USERINFO_ENCRYPTED_RESPONSE_ENC.toString()));
+        }
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUEST_OBJECT_SIGNING_ALG.toString())) &&
+                SignatureAlgorithm.fromString(respClaims.get(REQUEST_OBJECT_SIGNING_ALG.toString())) != null) {
+            rpFromRegisterResponse.setRequestObjectSigningAlg(respClaims.get(REQUEST_OBJECT_SIGNING_ALG.toString()));
+        }
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ALG.toString())) &&
+                KeyEncryptionAlgorithm.fromName(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ALG.toString())) != null) {
+            rpFromRegisterResponse.setRequestObjectEncryptionAlg(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ALG.toString()));
+        }
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString())) &&
+                BlockEncryptionAlgorithm.fromName(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString())) != null) {
+            rpFromRegisterResponse.setRequestObjectEncryptionEnc(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString()));
+        }
+
+        rpFromRegisterResponse.setDefaultMaxAge(safeToNumber(respClaims.get(DEFAULT_MAX_AGE.toString())));
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUIRE_AUTH_TIME.toString()))) {
+            rpFromRegisterResponse.setRequireAuthTime(Boolean.valueOf(respClaims.get(REQUIRE_AUTH_TIME.toString())));
+        }
+
+        rpFromRegisterResponse.setInitiateLoginUri(respClaims.get(INITIATE_LOGIN_URI.toString()));
+        rpFromRegisterResponse.setAuthorizedOrigins(convertStringToList(respClaims.get(AUTHORIZED_ORIGINS.toString())));
+        rpFromRegisterResponse.setAccessTokenLifetime(safeToNumber(respClaims.get(ACCESS_TOKEN_LIFETIME.toString())));
+        rpFromRegisterResponse.setSoftwareId(respClaims.get(SOFTWARE_ID.toString()));
+        rpFromRegisterResponse.setSoftwareVersion(respClaims.get(SOFTWARE_VERSION.toString()));
+        rpFromRegisterResponse.setSoftwareStatement(respClaims.get(SOFTWARE_STATEMENT.toString()));
+        rpFromRegisterResponse.setClientJwksUri(respClaims.get(JWKS_URI.toString()));
+        rpFromRegisterResponse.setClaimsRedirectUri(convertStringToList(respClaims.get(CLAIMS_REDIRECT_URIS.toString())));
+
+        return rpFromRegisterResponse;
+    }
+
+    private static List<String> convertStringToList(String input) {
```

Comment:
Can you check whether you can take advantage of existing `RegisterRequest.fromJson`?  Then we will not need to parse it manually. If not `new JSONObject(registerResponse.getEntity())` may look better. 

```
RegisterRequest parsed = RegisterRequest.fromJson(registerResponse.getEntity())

```

Reply:
Done. Used `RegisterRequest.fromJson`